applications/Tellor.md

```diff
@@ -0,0 +1,228 @@
+| **0a.** | License |  MIT|
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial**. |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| **0d.** | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 1 | Develop Controller contracts | We will provide a set of solidity smart contracts with the functionality described above in the [Project Details](#project-details)  |
+| 2 | Develop Parachain integration contract| We will provide an integration contract |
+
+
+### Milestone 2 — Create and test oracle pallet and complete documentation with usage examples
+
+- **Estimated Duration:** 2 months
+- **FTE:**  2
+- **Costs:** 20,000 USD
+
+Details:  A new Substrate pallet will be required which includes the core oracle data reporting and querying logic as well as staking rewards and payment logic, ported from the existing tellorFlex (oracle), AutoPay and UsingTellor (consumer) smart contracts. All the logic within the pallet will simply use the token of the parachain. Any parachain user can create and fund feeds, which are fulfilled by oracle reporters which have deposited a stake to the Oracle contract on the smart contract parachain.  Once completed, integrating and testing the pallet and the controller contracts will take significant work and documentation in order to make a robust and user-friendly oracle module.  
+
+| Number | Deliverable | Specification                                                                                                                                                   |
+| -----: | ----------- | ------------- |
+| **0a.** | License | MIT                                                                                                                                                             |
+| **0b.** | Documentation | We will provide both **inline documentation** of the code and a basic **tutorial**.                                                                             |
+| **0c.** | Testing and Testing Guide | Core functions will be fully covered by comprehensive unit tests to ensure functionality and robustness. In the guide, we will describe how to run these tests. |
+| **0d.** | Docker | We will provide a Dockerfile(s) that can be used to test all the functionality delivered with this milestone. |
+| 1 | Substrate Oracle pallet design and integration | We will provide the Substrate oracle pallet |
+| 2 | Testing functionallity of the the pallet and testing integration with a mock project and selected parachains| We will test full functionality of the system for interactions between the consumer chain and oracle pallet (meaning test the functinallity between milestone 1 and 2 -  solidity contracts, pallet, XCM)|
```

Comment:
Please specify what is being delivered here. A testing guide? Test report? Article?

Reply:
Hi @semuelle , thanks for the feedback.

> Please specify what is being delivered here. A testing guide? Test report? Article?

-Response: On deliverable 2, what is being delivered is just more testing. I read deliverable 0c as being more specific to the pallet testing (deliverable 1) so in deliverable 2 it would just be more test delivered/included that test the interoperability between milestone 1 and milestone 2 deliverable 1. I could remove it if you think that is covered by deliverable 0c.

Reply:
No, that's fine given the architecture. It just needs to be clear from the specification of the deliverable what exactly will be delivered, so that we know what to expect. You could add, for example, "we will provide a tutorial how to replicate these tests locally."

Reply:
Hi @semuelle , I've added clarification on milestone 2 deliverable 1 since we'll definitely provide a guide on running the additional tests.